config/stored_requests.go

```diff
@@ -190,12 +190,15 @@ func (cfg *DatabaseConfig) validate(dataType DataType, errs []error) []error {
 // DatabaseConnection has options which put types to the Database Connection string. See:
 // https://godoc.org/github.com/lib/pq#hdr-Connection_String_Parameters
 type DatabaseConnection struct {
-	Driver   string `mapstructure:"driver"`
-	Database string `mapstructure:"dbname"`
-	Host     string `mapstructure:"host"`
-	Port     int    `mapstructure:"port"`
-	Username string `mapstructure:"user"`
-	Password string `mapstructure:"password"`
+	Driver     string `mapstructure:"driver"`
+	Database   string `mapstructure:"dbname"`
+	Host       string `mapstructure:"host"`
+	Port       int    `mapstructure:"port"`
+	Username   string `mapstructure:"user"`
+	Password   string `mapstructure:"password"`
+	ServerCa   string `mapstructure:"server_ca"`
+	ClientCert string `mapstructure:"client_cert"`
+	ClientKey  string `mapstructure:"client_key"`
```

Comment:
Also, I just realized that we don't currently support TLS for postgres. I see that when we build the postgres connection string we disable SSL: `buffer.WriteString("sslmode=disable")`.
We should eventually add TLS support for postgres in another PR.

For now, I'd like us to carefully consider the config variable names here since they will ultimately be used for other database TLS configurations.
I'm looking at https://pkg.go.dev/github.com/lib/pq@v1.10.4 and I see `sslcert`, `sslkey` and `sslrootcert` which should correspond to what we have here now right?
As for the query param in the postgres connection string, I think we would just set that to one of the other values described in the docs but we would still need all of these certs configured here.

@VeronikaSolovei9 any preferences on naming?

Reply:
Good point! 
Hm, I would not combine all possible DB providers to the same config. So far we have generic host, post, user and password. Now we are about to add TLS support for MySql. For Postgres similar parameters are required, but have different names. 

Idea 1: 
Instead of these 3 variables add a `map[string]string` with extra DB parameters. It can lead to complications in setup, parsing, and figuring out what is what and for what DB provider. 

Idea 2:
Just add generic names like it defined in documentation: 
https://cloud.google.com/sql/docs/mysql/samples/cloud-sql-mysql-databasesql-connect-tcp-sslcerts

"DB_ROOT_CERT" // e.g., '/path/to/my/server-ca.pem'
"DB_CERT" // e.g. '/path/to/my/client-cert.pem'
"DB_KEY" // e.g. '/path/to/my/client-key.pem'

And rethink about it if something else will be needed in future.

Reply:
👍 idea 2 is what I had in mind as well, though I think we should use names that don't include `DB`, `SSL` or `TLS` in them since we should already have that context from the config structure: `stored_requests.database.connections.tls.VARIABLE`.

In that case, how about:
```
root_cert
client_cert
client_key
```

As for @holms's [request](https://github.com/prebid/prebid-server/pull/2507#discussion_r1081648100) regarding adding `InsecureSkipVerify` to the config, I think we can do that but we should consider whether it makes sense for this to add more generic variable names or use the `map[string]string` format presented as idea 1 for DB-specific parameters. I still prefer using generic names over a map for this as well if possible. For postgres there is a query param `sslmode` that can be set to one of six values, one of which might map to `InsecureSkipVerify`. @holms which, if any, of the postgres values sounds like it matches the `InsecureSkipVerify` behavior?
https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-PARAMKEYWORDS

Reply:
@bsardo  Perhaps `sslmode=verify-ca` would be the closest match? It doesn't check hostname, so probably self generated certs would pass with this param.

@victorshevtsov  I don't understand why there are three params in the config? You just need to provide a client key, which is pem certificate. I've just quickly googled how would I connect to mysql using tls with python for example, and all you need is just one certificate, also as digital ocean also provides only one certificate. Other params are specifically for mysql server, not client.

Reply:
> Also, I just realized that we don't currently support TLS for postgres. I see that when we build the postgres connection string we disable SSL: `buffer.WriteString("sslmode=disable")`. We should eventually add TLS support for postgres in another PR.
> 
> For now, I'd like us to carefully consider the config variable names here since they will ultimately be used for other database TLS configurations. I'm looking at https://pkg.go.dev/github.com/lib/pq@v1.10.4 and I see `sslcert`, `sslkey` and `sslrootcert` which should correspond to what we have here now right? As for the query param in the postgres connection string, I think we would just set that to one of the other values described in the docs but we would still need all of these certs configured here.
> 
> @VeronikaSolovei9 any preferences on naming?

Yeah I've actually planned to request Victor for another PR, so that TLS would be supported in postgesql too. Azure soon will be deprecating non-tls connections. Digital Ocean already doesn't allow to connect without tls to their postgresql managed service.

Reply:
1. Personally, I would go with the second idea, and use these names for the settings:
```
root_cert
client_cert
client_key
```

2. As @holms said, the client's cert and key parameters are optional. So, it is possible to pass only `root_cert` or all the 3 parameters.